### ISSUE TITLE
Default Branch Change

### ISSUE THREAD

**Issue by calebcartwright** [rust-lang/rustfmt#4801]
TL;DR - We're swapping the default repository branch so if you are currently compiling rustfmt from source to build the unreleased v2.0-rc, please be sure you are pointing explicitly to the `rustfmt-2.0.0-rc.2` [branch](https://github.com/rust-lang/rustfmt/tree/rustfmt-2.0.0-rc.2).

### What's changing
The default branch in this repository (`master`) has been used for the development of an experimental, unreleased major version (2.0) upgrade of rustfmt for some time, while development for the official/released/supported versions of rustfmt (1.x) occurs in feature branches. We're going to swap this back and change it so that the default/master branch in this repository contains the official v1.x version of rustfmt and have the 2.0 experimentation on a feature branch `rustfmt-2.0.0-rc.2`.

This change of the default branch will be made on, or shortly after, **April 25th** so if you are currently a consumer of the experimental 2.0 via a git reference please be sure to update any references to point to the `rustfmt-2.0.0-rc.2` branch to avoid any disruption.

### Why
It's been a source of confusion having the default branch in the repository diverge significantly from the official version of rustfmt that most users consume, as well as the version of the rustfmt submodule in the rust-lang/rust repository. Making this change will better reflect current state and should reduce some confusion and complexity.

### Rustfmt Developers/Contributors
This notification is primarily targeted to the relatively small audience of folks that are consumers of rustfmt by building the 2.0 version from source. There should be minimal impact to rustfmt developers but please feel free to ping me in any of your open PRs and/or on Discord if you have any questions.

### Off topic items
There's no information to share about a potential 2.0 release of rustfmt at this time. It's still experimental, and there's absolutely no guarantee it will ever be released, much less released as part of the Rust 1.x ecosystem.

**Comment by calebcartwright** [comment]
@dtolnay I believe you've mentioned before that you know some folks that are building v2 of rustfmt from source, so wanted to make sure you saw this

**Comment by doivosevic** [comment]
I just have to say I really love some of the options in 2.0. Some day ... :)
